tests/agent/test_agent_run_submission.py

```diff
@@ -1103,3 +1105,91 @@ def foo():
 
     submitted_flow_run_ids = {flow_run.id for flow_run in submitted_flow_runs}
     assert submitted_flow_run_ids == work_queue_flow_run_ids
+
+
+@pytest.fixture
+async def deployment_on_default_queue(
+    session,
+    flow,
+    flow_function,
+    infrastructure_document_id,
+    storage_document_id,
+    work_pool,
+):
+    def hello(name: str):
+        pass
+
+    deployment = await models.deployments.create_deployment(
+        session=session,
+        deployment=schemas.core.Deployment(
+            name="My High Priority Deployment",
+            tags=["test"],
+            flow_id=flow.id,
+            schedule=schemas.schedules.IntervalSchedule(
+                interval=datetime.timedelta(days=1),
+                anchor_date=pendulum.datetime(2020, 1, 1),
+            ),
+            storage_document_id=storage_document_id,
+            path="./subdir",
+            entrypoint="/file.py:flow",
+            infrastructure_document_id=infrastructure_document_id,
+            work_queue_name="wq",
+            parameter_openapi_schema=parameter_schema(hello),
+            work_queue_id=work_pool.default_queue_id,
+        ),
+    )
+    await session.commit()
+    return deployment
+
+
+async def test_agent_runs_high_priority_flow_runs_first(
+    orion_client: OrionClient,
+    deployment_in_non_default_work_pool: schemas.core.Deployment,
+    deployment_on_default_queue: schemas.core.Deployment,
+    work_pool: schemas.core.WorkPool,
+    work_queue_1: schemas.core.WorkQueue,
+):
```

Comment:
If at all possible it would be nice to have a docstring that explains what behavior we're seeing here and why. It feels like this test is going to be referenced frequently for explaining how priority and scheduling interact.